cpp/src/parquet/encoding.cc

```diff
@@ -2740,13 +2740,12 @@ class DeltaLengthByteArrayEncoder : public EncoderImpl,
       : EncoderImpl(descr, Encoding::DELTA_LENGTH_BYTE_ARRAY,
                     pool = ::arrow::default_memory_pool()),
         sink_(pool),
-        length_encoder_(nullptr, pool),
-        encoded_size_{0} {}
+        length_encoder_(nullptr, pool) {}
 
   std::shared_ptr<Buffer> FlushValues() override;
 
   int64_t EstimatedDataEncodedSize() override {
-    return encoded_size_ + length_encoder_.EstimatedDataEncodedSize();
+    return sink_.length() + length_encoder_.EstimatedDataEncodedSize();
```

Comment:
This will probably be slightly less precise (because `sink_.resize ` will resize to multiple of 64 bytes), but that's inline with expectations. I wonder if this would reduce number of allocations and improve performance?

Reply:
Would `length` and `capacity` be different, since "length" is only the logical size in `sink_`?

Reply:
Ah sorry in this case they wouldn't be different as we call `sink_.Reserve` (and not `sink_.Resize` as I assumed).